src/wsproxy.ml

```diff
@@ -35,27 +35,33 @@ let start path handler =
       (fun () ->
          Lwt_unix.accept fd_sock >>= fun (fd_sock2,_) ->
          let buffer = String.make 16384 '\000' in
-         let iov = Lwt_unix.io_vector ~buffer ~offset:0 ~length:16384 in
-         (
-           try Lwt_unix.recv_msg ~socket:fd_sock2 ~io_vectors:[iov]
-           with _ ->
-             Lwt_unix.close fd_sock2 >>= fun () ->
-             Lwt.return (0,[])
-         ) >>= fun (len,newfds) ->
-         let msg = String.sub buffer 0 len in
-         Lwt_unix.close fd_sock2 >>= fun _ ->
-         List.iter (fun fd -> Printf.printf "got fd: %d\n%!" (Obj.magic fd)) newfds;
-         Printf.printf "About to fixup the fd\n%!";
+         Lwt.catch
+           (fun () ->
+              let iov = Lwt_unix.io_vector ~buffer ~offset:0 ~length:16384 in
+              Lwt_unix.recv_msg ~socket:fd_sock2 ~io_vectors:[iov])
+           (fun e ->
+              Printf.printf "Caught exception: %s\n" (Printexc.to_string e);
+              ignore_exn (fun () -> Lwt_unix.close fd_sock2) () >>= fun () ->
+              Lwt.return (0,[]))
+         >>= fun (len,newfds) ->
+         Lwt_unix.close fd_sock2 >>= fun () ->
          Lwt.catch
            (fun () ->
               let fdh :: fdt = newfds in
-              ignore(handler (Lwt_unix.of_unix_file_descr fdh) msg);
+              let msg = String.sub buffer 0 len in
+              let fdno = Printf.sprintf "%d" (Obj.magic fdh) in
+              List.iter (fun fd -> Printf.printf "closing fds: %d\n%!" (Obj.magic fd)) fdt;
               List.iter (fun fd -> try Unix.close fd with _ -> ()) fdt;
```

Comment:
I assume it's not a disaster that we're not using `Lwt_unix.close` here. But we should fix this in a subsequent PR.